tools/c7n_azure/c7n_azure/resources/network_security_group.py

```diff
@@ -164,19 +164,20 @@ def process(self, network_security_groups, event=None):
         self.IsAllowed = StringUtils.equal(self.data.get(ACCESS), ALLOW_OPERATION)
         self.match = self.data.get(MATCH, 'all')
 
-        # Calculate ports from the settings:
-        #   If ports not specified -- assuming the entire range
-        #   If except_ports not specifed -- nothing
-        ports_set = PortsRangeHelper.get_ports_set_from_string(self.data.get(PORTS, '0-65535'))
-        except_set = PortsRangeHelper.get_ports_set_from_string(self.data.get(EXCEPT_PORTS, ''))
-        self.ports = ports_set.difference(except_set)
-        self.source_address = self.data.get(SOURCE, None)
-        self.destination_address = self.data.get(DESTINATION, None)
+        if ports := self.data.get(PORTS):
+            # Calculate ports from the settings
+            ports_set = PortsRangeHelper.get_ports_set_from_string(ports)
+            except_set = PortsRangeHelper.get_ports_set_from_string(self.data.get(EXCEPT_PORTS, ''))
+            self.ports = ports_set.difference(except_set)
+
+            self.source_address = self.data.get(SOURCE, None)
+            self.destination_address = self.data.get(DESTINATION, None)
```

Comment:
This is mostly indentation. We don't use any of these attributes if no port is set on the filter, so there's no need to set them. It also helps protect against running `check_nsg` with incomplete info.